packages/webpack-cli/lib/plugins/InteractiveModePlugin.js

```diff
@@ -0,0 +1,201 @@
+const readline = require("readline");
+const { red, green, cyanBright, bold, gray } = require("colorette");
+const { SyncHook } = require("tapable");
+const logger = require("../utils/logger");
+
+/**
+ * Displays command space at bottom of screen
+ * @param {string} msg message to print with command
+ * @param {boolean} status currently watching or not
+ */
+const spawnCommand = (msg, status, toClear = false, verbose = false) => {
+    const lines = verbose ? 3 : 2;
+    const totalRows = process.stdout.rows;
+    readline.cursorTo(process.stdout, 0, totalRows - lines);
+    readline.clearScreenDown(process.stdout);
+
+    if (toClear) return;
+
+    if (verbose) {
+        process.stdout.write(`${bold(cyanBright(`i  ${msg}`))}\n`);
+        process.stdout.write(`${gray("   q: quit  w: watch  s: pause")}`);
+    } else {
+        process.stdout.write("\n");
+    }
+
+    readline.cursorTo(process.stdout, 0, totalRows - 2);
+
+    // for current status
+    if (status) {
+        process.stdout.write(`${green("▲")}  `);
+    } else {
+        process.stdout.write(`${red("▲")}  `);
+    }
+
+    readline.cursorTo(process.stdout, 0, totalRows - lines - 1);
+};
+
+/**
+ * Clear the whole terminal
+ */
+const clrscr = () => {
+    process.stdout.write("\x1B[2J\x1B[3J\x1B[H");
+};
+
+/**
+ * Interactive Mode plugin
+ */
+class InteractiveModePlugin {
+    constructor(mode = "verbose") {
+        this.name = "webpack-cli-interactive-mode";
+        this.keys = {
+            quit: "q",
+            stop: "s",
+            start: "w",
+        };
+        this.handlers = {
+            quit: this.quitHandler.bind(this),
+            stop: this.stopHandler.bind(this),
+            start: this.startHandler.bind(this),
+        };
+        this.logger = undefined;
+        this.verbose = mode === "verbose";
+
+        // hide cursor
+        process.stdout.write("\u001B[?25l");
```

Comment:
Avoid doing something extra in configuration, move it to apply

Reply:
Done.